### PR TITLE
sketchybar: 2.8.2 -> 2.12.3

### PR DESCRIPTION
**Pull request by azuwis**
###### Description of changes

Replace https://github.com/NixOS/nixpkgs/pull/203897

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by luxus** [comment]
Result of `nixpkgs-review pr 209260` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sketchybar</li>
  </ul>
</details>

**Comment by luxus** [comment]
maybe this will fix some other building problems as well.. good catch @azuwis ..

**Comment by luxus** [comment]
@azuwis if you have time, it would be great to change the PR to the latest version

**Comment by luxus** [comment]
https://github.com/FelixKratz/SketchyBar/releases/tag/v2.13.2 could you update to the release?

**Comment by luxus** [comment]
here is a patch for the newest version
```patch
diff --git a/pkgs/os-specific/darwin/sketchybar/default.nix b/pkgs/os-specific/darwin/sketchybar/default.nix
index 86d3206..fff6a66 100644
--- a/pkgs/os-specific/darwin/sketchybar/default.nix
+++ b/pkgs/os-specific/darwin/sketchybar/default.nix
@@ -10,13 +10,13 @@ in
 
 stdenv.mkDerivation rec {
   pname = "sketchybar";
-  version = "2.12.3";
+  version = "2.13.2";
 
   src = fetchFromGitHub {
     owner = "FelixKratz";
     repo = "SketchyBar";
     rev = "v${version}";
-    sha256 = "sha256-ZTEJvwJVUq1tQY/bFU209oMIykKI4Q0A8ozrq3JBdEY=";
+    sha256 = "sha256-EPpD1QuJXFFwWcs1gmhjf9zm6mkEM72Zr8NOAAgxI4I=";
   };
 
   buildInputs = [ Carbon Cocoa DisplayServices SkyLight ]


```

File: pkgs/os-specific/darwin/apple-sdk/frameworks.nix [hunk 1]
```diff
@@ -43,6 +43,7 @@ with frameworks; with libs; {
   DiscRecording           = { inherit libobjc CoreServices IOKit; };
   DiscRecordingUI         = {};
   DiskArbitration         = { inherit IOKit; };
+  DisplayServices         = {};
```
Comment (wegank) [inline pkgs/os-specific/darwin/apple-sdk/frameworks.nix:46]: This is not necessary.

File: pkgs/os-specific/darwin/apple-sdk/impure-deps.nix [hunk 2]
```diff
@@ -191,6 +191,9 @@ rec {
   CoreData = [
     "/System/Library/Frameworks/CoreData.framework"
   ];
+  DisplayServices = [
+    "/System/Library/PrivateFrameworks/DisplayServices.framework/Versions/A/DisplayServices"
+  ];
```
Comment (wegank) [inline pkgs/os-specific/darwin/apple-sdk/impure-deps.nix:196]: Nor this.

File: pkgs/os-specific/darwin/sketchybar/default.nix [hunk 3]
```diff
@@ -10,16 +10,16 @@ in
...
     owner = "FelixKratz";
     repo = "SketchyBar";
     rev = "v${version}";
-    sha256 = "sha256-GmM+0h6xxUzW2kpTDZWAiqJAXoQgdsJRlNbvsuxKmZ8=";
+    sha256 = "sha256-EPpD1QuJXFFwWcs1gmhjf9zm6mkEM72Zr8NOAAgxI4I=";
   };
 
-  buildInputs = [ Carbon Cocoa SkyLight ]
+  buildInputs = [ Carbon Cocoa DisplayServices SkyLight ]
     ++ lib.optionals (stdenv.system == "x86_64-darwin") [ memstreamHook ];
```
Comment (wegank) [inline pkgs/os-specific/darwin/sketchybar/default.nix:23]: Is this still necessary when using `darwin.apple_sdk_11_0.callPackage`?

**Review by wegank — APPROVED** [review]

**Comment by luxus** [comment]
@wegank thanks for reviewing it, i guess many of the mistakes have been done by me.

azuwis, thanks for fixing everything. 
OfBorg is happy as well and it should be ready to merged

**Pull request merged**